module/scripts/combat/effects.js

```diff
@@ -0,0 +1,94 @@
+
+async function changeHealth(actor, amount, name, remidy = "") {
+    // Ensure `amount` is a number
+    if (typeof amount !== 'number' || !isFinite(amount)) {
+      console.error(`Invalid amount for health change: ${amount}`);
+      return;
+    }
+  
+    const currentHP = actor.system.derivedStats.hp.value;
+    const maxHP = actor.system.derivedStats.hp.max;
+    const newHP = Math.min(Math.max(currentHP + amount, 0), maxHP);
+  
+    // Ensure the new HP is within valid range
+    if (newHP !== currentHP) {
+      await actor.update({
+        'system.derivedStats.hp.value': newHP
+      });
+  
+      if (amount > 0) {
+        ChatMessage.create({
+          content: `${actor.name} heals for ${amount} hit points because of ${name}!${remidy ? ` ${remidy}` : ''}`,
+          speaker: ChatMessage.getSpeaker({ actor: actor })
+        });
+      } else if (amount < 0) {
+        ChatMessage.create({
+          content: `${actor.name} takes ${-amount} hit points of ${name} damage!`,
+          speaker: ChatMessage.getSpeaker({ actor: actor })
+        });
+      }
+    } else {
+      console.log(`No change in HP detected for ${actor.name}.`);
+    }
+  }
+  
+  async function applyFireDamageToEquippedItems(actor, damage) {
+    // Update stopping power for equipped armor
+    for (const item of actor.items.filter(i => i.type === 'armor' && i.system.equipped)) {
+        const locationsCovered = ['head', 'torso', 'leftArm', 'rightArm', 'leftLeg', 'rightLeg'];
+        for (const location of locationsCovered) {
+            const stoppingField = `${location}Stopping`;
+            if (item.system[stoppingField] !== undefined) {
+                const stoppingPower = item.system[stoppingField] ?? 0;
+                if (stoppingPower > 0) {
+                    const newStoppingPower = Math.max(stoppingPower - damage, 0);
+                    await item.update({ [`system.${stoppingField}`]: newStoppingPower });
+                    ui.notifications.info(`${item.name} loses ${damage} stopping power in ${location}!`,
+                        { permanent: false, delay: 1000 });
+                }
+            }
+        }
+    }
+
+    // Update reliability for equipped weapons
+    for (const item of actor.items.filter(i => i.type === 'weapon' && i.system.equipped)) {
+        const reliabilityValue = item.system?.reliable ?? 0;
+        const newReliability = Math.max(reliabilityValue - 1, 0);
+        await item.update({ "system.reliable": newReliability });
+        ui.notifications.info(`${item.name} loses 1 reliability!`, { permanent: false, delay: 1000 });
+    }
+}
+
+async function applyFire(actor) {
+    const fireDamage = -5;
+    // Add fire damage calculation
+    ChatMessage.create({
+        content: `${actor.name} burns for a total of 5 damage to all locations (blocked by armor).
+            Please calculate manually! All equipped armor and weapons got damaged.`,
+        speaker: ChatMessage.getSpeaker({ actor: actor })
```

Comment:
There is a method to apply damage to all locations

Reply:
As mentioned above that function does not allow me to control SP damage.

Reply:
Using now the applyNormalDamage function for all effects. I would be not sure about the healing. Where would that go?

Reply:
https://github.com/witchertrpg-foundryvtt/TheWitcherTRPG/blob/main/module/scripts/chat.js#L59

you need to split it so that you have a method that accepts an actor and the heal and the current method just extract these values from the event and calls it.

How this should be structured you can see here: https://github.com/Stexinator/salvage-union-combat-automation/blob/main/src/scripts/weapons.js#L55